src/main/src/main/java/org/geoserver/config/GeoServerPersister.java

```diff
@@ -576,23 +583,53 @@ private void addStyle( StyleInfo s ) throws IOException {
     
     private void renameStyle( StyleInfo s, String newName ) throws IOException {
         LOGGER.fine( "Renameing style " + s.getName() + " to " + newName );
+        
+        // rename xml configuration file
         Resource xml = dd.config(s);
         renameRes( xml, newName+".xml" );
         
+        // rename style definition file
         Resource style = dd.style(s);
-        String sldFileName = newName + ".sld";
-        Resource target = style.parent().get(sldFileName);
-        int i = 1;
-        while(target.getType()!=Type.UNDEFINED && i <= MAX_RENAME_ATTEMPTS) {
-            sldFileName = newName + i + ".sld";
-            target = style.parent().get(sldFileName);
+        StyleHandler format = Styles.handler( s.getFormat() );
+        
+        Resource target = uniqueResource( style, newName, format.getFileExtension() );
+        renameRes(style, target.name());
+        s.setFilename(target.name());
+        
+        // rename generated sld if appropriate
+        if( !"sld".equals(format.getFormat())){
```

Comment:
Using `format.getFileExtension()` above, but `format.getFormat()` here?
Should use `SLDHandler.FORMAT.equals(format.getFormat())) ` instead.
Alternatively, `"sld".equalsIgnoreCase(format.getFileExtension())`